src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.cpp

```diff
@@ -161,4 +162,23 @@ ep_rt_coreclr_sample_profiler_write_sampling_event_for_threads (
 	return;
 }
 
+void
+ep_rt_coreclr_init_finish (void)
+{
+        CONTRACTL
+        {
+                THROWS;
+                GC_TRIGGERS;
+                MODE_ANY;
+                INJECT_FAULT(COMPlusThrowOM(););
+        }
+        CONTRACTL_END;
+
+        GCX_COOP();
+
+        // FIXME: how do I check if the method was trimmed away and avoid calling it?
+        MethodDescCallSite runtimeEventSourceInitialize(METHOD__RUNTIME_EVENT_SOURCE__INITIALIZE);
+        runtimeEventSourceInitialize.Call(NULL);
```

Comment:
Question for CLR folks: I put `DEFINE_CLASS(RUNTIME_EVENT_SOURCE, ...)` wrapped in a `BEGIN_ILLINK_FEATURE_SWITCH` which means the method may be trimmed away if eventpipe is disabled.  How do I actually make this `MethodDescCallSite` lookup allowed to fail and avoid doing a call?

Or is this baking a method token right into the runtime?

Reply:
In other places that I know of we don't rely on the lookup failing, instead we react to the feature switch in question and avoid doing the lookup completely. (For example built-in COM works this way)

Reply:
how can I check the value of the feature switch at runtime from native?

Reply:
I guess I can do `DEFINE_STATIC_PROPERTY` for `IsSupported` and call that and then call the Initialize method